pkg/util/compat/json_slice_merge.go

```diff
@@ -0,0 +1,59 @@
+package compat
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+)
+
+// JSONSliceMerge will merge two slices using JSONCopy according to these rules:
+//
+// 1. If `dst` and `src` are the same length, all elements are merged
+//
+// 2. If `dst` is longer, only the first `len(src)` elements are merged
+//
+// 3. If `src` is longer, first `len(dst)` elements are merged, then remaining elements are appended to `dst`
+func JSONSliceMerge(dst, src interface{}) error {
+	dstVal := reflect.ValueOf(dst)
+	srcVal := reflect.ValueOf(src)
+
+	if dstVal.Kind() != reflect.Ptr {
+		return errors.New("dst must be a pointer to slice")
+	}
+
+	dstVal = reflect.Indirect(dstVal)
+	srcVal = reflect.Indirect(srcVal)
+
+	if dstVal.Kind() != reflect.Slice {
+		return errors.New("dst must be pointing to a slice")
+	}
+
+	if srcVal.Kind() != reflect.Slice {
+		return errors.New("src must be a slice or a pointer to slice")
+	}
+
+	minLen := dstVal.Len()
+	if srcVal.Len() < minLen {
+		minLen = srcVal.Len()
+	}
+
+	// merge common elements
+	for i := 0; i < minLen; i++ {
+		dstX := dstVal.Index(i).Addr().Interface()
```

Comment:
[q] am I right that this will work only if the items of the slice are pointers as well? (looking at the tests below seems only the destination slice must have pointers?)
If so - may need to add to the method documentation

Reply:
They don't have to be pointers - I used pointers in the test case just to check that pointers work too